src/client/java/teammates/client/scripts/DataMigrationForSanitizedDataInStudentProfileAttributes.java

```diff
@@ -0,0 +1,109 @@
+    /**
+     * Will not perform updates on the datastore if true.
+     */
+    private static final boolean isPreview = true;
+    private ProfilesLogic profilesLogic = ProfilesLogic.inst();
+
+    public static void main(String[] args) throws IOException {
+        new DataMigrationForSanitizedDataInStudentProfileAttributes().doOperationRemotely();
+    }
+
+    @Override
+    protected void doOperation() {
+        List<StudentProfileAttributes> allStudentProfiles = profilesLogic.getAllStudentProfiles();
+        int numberOfAffectedProfiles = 0;
+        int numberOfUpdatedProfiles = 0;
+        LoopHelper loopHelper = new LoopHelper(100, "student profiles processed.");
+        println("Running data migration for sanitization on student profiles...");
+        println("Preview: " + isPreview);
+        for (StudentProfileAttributes profile : allStudentProfiles) {
+            loopHelper.recordLoop();
+            boolean isProfileUnsanitized = !isProfileSanitized(profile);
+            if (isProfileUnsanitized) {
+                // skip the update if profile is already unsanitized
+                continue;
+            }
+            // profile contains sanitized fields
+            numberOfAffectedProfiles++;
+            try {
+                desanitizeAndUpdateProfile(profile);
+                numberOfUpdatedProfiles++;
+            } catch (InvalidParametersException | EntityDoesNotExistException e) {
+                println("Problem sanitizing profile with google id " + profile.googleId);
+                println(e.getMessage());
+            }
+        }
+        println("Total number of profiles: " + loopHelper.getCount());
+        println("Number of affected profiles: " + numberOfAffectedProfiles);
+        println("Number of updated profiles: " + numberOfUpdatedProfiles);
+    }
+
+    /**
+     * Desanitizes the fields of {@code profile} and updates it in the database.
+     */
+    private void desanitizeAndUpdateProfile(StudentProfileAttributes profile)
+            throws InvalidParametersException, EntityDoesNotExistException {
+
+        profile.shortName = getDesanitizedIfSanitized(profile.shortName);
+        profile.email = getDesanitizedIfSanitized(profile.email);
+        profile.institute = getDesanitizedIfSanitized(profile.institute);
+        profile.nationality = getDesanitizedIfSanitized(profile.nationality);
+        profile.gender = getDesanitizedIfSanitized(profile.gender);
+        profile.moreInfo = getDesanitizedIfSanitized(profile.moreInfo);
+
+        if (!profile.isValid()) {
+            throw new InvalidParametersException(profile.getInvalidityInfo());
+        }
+
+        if (!isPreview) {
+            profilesLogic.updateStudentProfile(profile);
+        }
+    }
+
+    /**
+     * Returns true if any field in {@code profile} is sanitized.
+     */
+    private boolean isProfileSanitized(StudentProfileAttributes profile) {
```

Comment:
Do you mean that according to the name`isProfileSanitized`, it should only return true if all fields are sanitized? Hmm I feel that this naming is the most readable though, and the logic written like this is easier to understand compared to negating it, as it will contain less `!` and there isn't a negation in the name.